dspnode/src/main/java/io/coti/dspnode/services/TransactionService.java

```diff
@@ -99,6 +150,51 @@ protected void handlePostponedTransaction(TransactionData postponedTransaction,
         }
     }
 
+    @Scheduled(initialDelay = 1000, fixedDelay = 86400000)
+    private void clearRejectedTransactions() {
+        rejectedTransactions.forEach(rejectedTransaction -> {
+                    if (rejectedTransaction != null && (Instant.now().getEpochSecond() - rejectedTransaction.getRejectionTime().getEpochSecond() > REJECTED_TRANSACTIONS_TTL)) {
+                        log.debug("removing rejected transaction due to TTL. hash: {}, rejection time: {}, reason: {}",
+                                rejectedTransaction.getHash(), rejectedTransaction.getRejectionTime(), rejectedTransaction.getRejectionReasonDescription());
+                        rejectedTransactions.delete(rejectedTransaction);
+                    }
+                }
+        );
+    }
+
+    private void processRejectedTransactions(TransactionData rejectedTransaction) {
+        if (rejectedTransactions.getByHash(rejectedTransaction.getHash()) != null && postponedTransactionMap.size() > 0) {
+            Map<TransactionData, Boolean> postponedChildrenTransactionMap = getPostponedChildrenTransactionMap(rejectedTransaction.getHash(), postponedTransactionMap);
+            for (Map.Entry<TransactionData, Boolean> entry : postponedChildrenTransactionMap.entrySet()) {
+                if (entry.getValue().equals(true)) {
+                    handleRejectedTransaction(entry.getKey(), REJECTED_PARENT);
+                    postponedTransactionMap.remove(entry.getKey());
+                    processRejectedTransactions(entry.getKey());
+                }
+            }
+        }
+    }
+
+    public ResponseEntity<IResponse> getRejectedTransactions() {
+        try {
+            List<RejectedTransactionResponseData> rejectedTransactionDataList = new ArrayList<>();
+            rejectedTransactions.forEach(rejectedTransaction -> rejectedTransactionDataList.add(new RejectedTransactionResponseData(rejectedTransaction)));
+            return ResponseEntity.ok(new GetRejectedTransactionsResponse(rejectedTransactionDataList));
+        } catch (Exception e) {
+            log.info("Exception while getting rejected transactions", e);
+            return ResponseEntity
+                    .status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(new Response(
+                            TRANSACTION_NONE_INDEXED_SERVER_ERROR,
+                            STATUS_ERROR));
+        }
+    }
+
+    @Override
+    public long getRejectedTransactionsSize() {
+        return rejectedTransactions.size();
```

Comment:
we are doing a infrastracture method in a transaction service. maybe we can move this to the collection it self?

Reply:
It is interface method, should be implemented differently in each module